src/engine/GroupBy.cpp

```diff
@@ -779,24 +785,23 @@ bool GroupBy::isUnsupportedAggregate(sparqlExpression::SparqlExpression* expr) {
   if (!expr->isAggregate()) return false;
 
   // `expr` is an unsupported aggregate
-  if (hasType<SumExpression>(expr) || hasType<MinExpression>(expr) ||
-      hasType<MaxExpression>(expr) || hasType<CountExpression>(expr) ||
-      hasType<GroupConcatExpression>(expr))
+  if (hasAnyType<SumExpression, MinExpression, MaxExpression, CountExpression,
+                 GroupConcatExpression>(expr))
     return true;
 
-  return false;
+  // `expr` is a distinct aggregate
+  return expr->isDistinct();
```

Comment:
I see, that is much simpler than what I had in mind, and adding the `isDistinct()` to all expressions doesn't really hurt:)

Reply:
And that is now a simple function that is easy to read:) Nice.